tests/e2e/mixedos/mixedos_test.go

```diff
@@ -21,6 +21,12 @@ var windowsAgentCount = flag.Int("windowsAgentCount", 1, "number of windows agen
 
 const defaultWindowsOS = "jborean93/WindowsServer2022"
 
+// getLabelPodIPs returns the IPs of the pods with a label
+func getLabelPodIPs(kubeConfigFile string, label string) ([]e2e.ObjIP, error) {
+	cmd := `kubectl get pods -l ` + label + ` -o=jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.podIPs[*].ip}{"\n"}{end}' --kubeconfig=` + kubeConfigFile
+	return e2e.GetObjIPs(cmd)
```

Comment:
If feels really weird to have the `GetObjIPs` function here that expects the specific output of this jsonpath expression. Can we maybe move this to testutils and structure it a bit more like `FetchClusterIP` and other commands that are responsible for both building the command and parsing the output?

Reply:
you are right, I was copy/pasting the k3s stuff. Let me refactor it

Reply:
It kinda feels like at some point, once you start getting super fancy with kubectl commands and trying to extract nested fields, you might as well just have the tests use the proper Kubernetes golang client libs instead of wrapping kubectl.